src/libraries/System.Data.Common/src/System/Xml/XPathNodePointer.cs

```diff
@@ -67,9 +64,10 @@ private XPathNodeType DecideXPNodeTypeForTextNodes(XmlNode node)
             //the function can only be called on text like nodes.
             Debug.Assert(XmlDataDocument.IsTextNode(node.NodeType));
             XPathNodeType xnt = XPathNodeType.Whitespace;
-            while (node != null)
+            XmlNode? n = null;
```

Comment:
Why change `node` to `n` here? I think we should revert this

Reply:
Because `node` is  parameter, which is non-nullable. The code assigns to the parameter, which then can include `null` values as the loop goes. And the assert would null ref when one passes null :-(